lib/spack/spack/build_systems/README-intel.rst

```diff
@@ -0,0 +1,522 @@
+                       library): i.e. -profile=myprofile for the myprofile.cfg file.
+                       As a special case, lib<name>.so or lib<name>.a may be used
+                       if the library is found
+       -check_mpi      link against the Intel(R) Trace Collector (-profile=vtmc).
+       -static_mpi     link the Intel(R) MPI Library statically
+       -mt_mpi         link the thread safe version of the Intel(R) MPI Library
+       -ilp64          link the ILP64 support of the Intel(R) MPI Library
+       -no_ilp64       disable ILP64 support explicitly
+       -fast           the same as -static_mpi + pass -fast option to a compiler.
+       -t or -trace
+                       link against the Intel(R) Trace Collector
+       -trace-imbalance
+                       link against the Intel(R) Trace Collector imbalance library
+                       (-profile=vtim)
+       -dynamic_log    link against the Intel(R) Trace Collector dynamically
+       -static         use static linkage method
+       -nostrip        turn off the debug information stripping during static linking
+       -O              enable optimization
+       -link_mpi=<name>
+                       link against the specified version of the Intel(R) MPI Library
+    All other options will be passed to the compiler without changing.
+    ----------------------------------------------------------------------------
+    The following environment variables are used:
+       I_MPI_ROOT      the Intel(R) MPI Library installation directory path
+       I_MPI_F90 or MPICH_F90
+                       the path/name of the underlying compiler to be used
+       I_MPI_FC_PROFILE or I_MPI_F90_PROFILE or MPIF90_PROFILE
+                       the name of profile file (without extension)
+       I_MPI_COMPILER_CONFIG_DIR
+                       the folder which contains configuration files *.conf
+       I_MPI_TRACE_PROFILE
+                       specify a default profile for the -trace option
+       I_MPI_CHECK_PROFILE
+                       specify a default profile for the -check_mpi option
+       I_MPI_CHECK_COMPILER
+                       enable compiler setup checks
+       I_MPI_LINK      specify the version of the Intel(R) MPI Library
+       I_MPI_DEBUG_INFO_STRIP
+                       turn on/off the debug information stripping during static linking
+       I_MPI_FCFLAGS
+                       special flags needed for compilation
+       I_MPI_LDFLAGS 
+                       special flags needed for linking
+    ----------------------------------------------------------------------------
+
+
+**************
+macOS support
+**************
+
+- On macOS, the Spack methods here only include support to integrate an
+  externally installed MKL.
+
+- URLs in child packages will be Linux-specific; macOS download packages
+  are located in differently numbered dirs and are named m_*.dmg.
```

Comment:
@adamjstewart wrote:
> Your comments ask about Intel's support for other platforms beside Linux? See #7403.

Yep. Thank you – I noticed it only briefly a few days ago and didn't know yet what to make of it. I now see that #7403 concerns the _installation_ phase only, retaining runtime support (as far as possible) of Spack-external Intel packages.

Reply:
Oh, true. Yes, if someone wants to use the package on macOS, they should be able to install it by hand, and then add it as an external package. After that, Spack should just work. Can anyone install Intel on macOS and take a look at the installation directory hierarchy to see if it matches?

Reply:
> Can anyone install Intel on macOS and take a look at the installation directory hierarchy to see if it matches?

I already did it a while ago, that's why we have all those `if platform == darwin` in `intel-mkl`.

Reply:
p.s. it does not match

Reply:
Indeed – I tried to keep @davydden’s dir selections intact and need to verify that I did them justice. I just today installed MKL on my box for that.

I found that the installer is non-native and unsurprisingly has a permissions quirk (not relevant for Spack directly, but could be frustrating for users). I commented to that end at #7403 in the lib/spack/docs/known_issues.rst changes (tagged outdated now).